catalog/dags/providers/provider_api_scripts/auckland_museum.py

```diff
@@ -0,0 +1,206 @@
+"""
+Content Provider:       Auckland War Memorial Museum Tāmaki Paenga Hira
+
+ETL Process:            Use the API to identify all CC licensed media.
+
+Output:                 TSV file containing the media and the
+                        respective meta-data.
+
+Notes:                  https://api.aucklandmuseum.com/
+
+Resource:               https://api.aucklandmuseum.com/
+                        https://github.com/AucklandMuseum/API/wiki/Tutorial
+
+Resource | Requests per second | Requests per day
+-- | -- | --
+/search, /id | 10 | 1000
+/id/media | 10 | 1000
+"""
+import logging
+from datetime import datetime, timedelta
+
+from common.constants import IMAGE
+from common.licenses import get_license_info
+from common.loader import provider_details as prov
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
+
+
+def convert_date_format(date_obj) -> str:
+    date = str(date_obj)
+    date = date.replace(" ", "T")
+    date = date + "Z"
+    return date
+
+
+logger = logging.getLogger(__name__)
+
+LANDING_URL = (
+    "https://www.aucklandmuseum.com/collections-research/collections/record/am_"
+)
+
+
+class AucklandMuseumDataIngester(ProviderDataIngester):
+    providers = {
+        "image": prov.AUCKLAND_MUSEUM_IMAGE_PROVIDER,
+    }
+    endpoint = "https://api.aucklandmuseum.com/search/collectionsonline/_search"
+    license_url = "https://creativecommons.org/licenses/by/4.0/"
+    total_amount_of_data = 10000
+    DEFAULT_LICENSE_INFO = get_license_info(license_url=license_url)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.delay = 4
+        self.batch_start = 0
+        self.batch_limit = 2000
+        self.headers = {"Content-Type": "application/json"}
+        self.date_from = convert_date_format(datetime.now() - timedelta(days=1))
+        self.date_to = convert_date_format(datetime.now())
```

Comment:
Instead of basing these on `datetime.now()`, we should use the logical date of the DagRun, which you can get using `self.date` (see Metropolitan for example). We should use `self.date` as the `date_from`, and `self.date + timedelta(days=1)` for `date_to`.

This would allow us to do backfills, for example -- if you run the Dag with `date` set to `10-01-23`, it will ingest records for that date, instead of re-ingesting today's records.

Reply:
That is a great idea!